pkg/cgroup/cgroup.go

```diff
@@ -0,0 +1,292 @@
+package cgroup
+
+import (
+	"bufio"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/aquasecurity/tracee/pkg/mount"
+)
+
+type Cgroup interface {
+	initialize() error
+	Destroy() error
+	GetMountPoint() string
+	GetDefaultHierarchy() int
+	IsV1() bool
+}
+
+// cgroupv1
+
+func InitCgroupV1() (Cgroup, error) {
+	c := &cgroupV1{}
+	err := c.initialize()
+	return c, err
+}
+
+type cgroupV1 struct {
+	mountPath string
+	hierarchy int
+}
+
+func (c *cgroupV1) initialize() error {
+
+	path, err := os.MkdirTemp(TmpCgroup, TmpCgroupV1Path)
+	if err != nil {
+		return failCreateTempDir(1, err)
+	}
+
+	mountPath, err := filepath.Abs(path)
+	if err != nil {
+		return failGetAbsPath(1, path)
+	}
+
+	mp, _ := mount.IsMountpointOf(mountPath, CgroupV1FsType)
+	if mp {
+		return cgroupAlreadyMounted(1, mountPath)
+	}
+
+	// mount -t cgroup -o cpuset cgroup mountPath
+	err = syscall.Mount(CgroupV1FsType, mountPath, CgroupV1FsType, 0, CgroupV1Controller)
+	if err != nil {
+		return failMountCgroup(1, err)
+	}
+
+	c.mountPath = mountPath
+
+	c.hierarchy, err = GetCgroupControllerHierarchy("cpuset")
+	if err != nil {
+		return failGetCgroupHierarchy(1, err)
+	}
+
+	return nil
+}
+
+func (c *cgroupV1) Destroy() error {
+
+	err := syscall.Unmount(c.mountPath, 0)
+	if err != nil {
+		return err
+	}
+	err = os.RemoveAll(c.mountPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *cgroupV1) GetMountPoint() string {
+	return c.mountPath
+}
+
+func (c *cgroupV1) GetDefaultHierarchy() int {
+	return c.hierarchy
+}
+
+func (c *cgroupV1) IsV1() bool {
+	return true
+}
+
+// cgroupv2
+
+func InitCgroupV2() (Cgroup, error) {
+	c := &cgroupV2{}
+	err := c.initialize()
+	return c, err
+}
+
+type cgroupV2 struct {
+	mountPath string
+	hierarchy int
+}
+
+func (c *cgroupV2) initialize() error {
```

Comment:
This function looks the same as the cgroup v1 initialize function. I think it can be generalized to take relevant parameters and handle both to avoid code duplication